search-website-functions-v4/api/Lookup.cs

```diff
@@ -0,0 +1,56 @@
+using System.Net;
+using Azure.Search.Documents.Models;
+using Azure.Search.Documents;
+using Azure;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Http;
+using Microsoft.Extensions.Logging;
+using WebSearch.Models;
+using Azure.Core;
+
+namespace WebSearch.Function
+{
+    public class Lookup
+    {
+        private static string searchApiKey = Environment.GetEnvironmentVariable("SearchApiKey", EnvironmentVariableTarget.Process);
+        private static string searchServiceName = Environment.GetEnvironmentVariable("SearchServiceName", EnvironmentVariableTarget.Process);
+        private static string searchIndexName = Environment.GetEnvironmentVariable("SearchIndexName", EnvironmentVariableTarget.Process) ?? "good-books";
+
+        private readonly ILogger _logger;
+
+        public Lookup(ILoggerFactory loggerFactory)
+        {
+            _logger = loggerFactory.CreateLogger<Lookup>();
+        }
+
+        [Function("lookup")]
+        public async Task<HttpResponseData> RunAsync([HttpTrigger(AuthorizationLevel.Anonymous, "get", "post")] HttpRequestData req, FunctionContext executionContext)
+        {
+
+            // Get Document Id
+            var query = System.Web.HttpUtility.ParseQueryString(req.Url.Query);
+            string documentId = query["id"].ToString();
+
+            // Cognitive Search 
+            Uri serviceEndpoint = new Uri($"https://{searchServiceName}.search.windows.net/");

```

Comment:
```suggestion
            Uri serviceEndpoint = new($"https://{searchServiceName}.search.windows.net/");
```

When the target type that you're instantiating is known (the declaration explicitly defines its type, instead of `var`) you should use target type new expressions for simplicity. https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-9.0/target-typed-new